File: bff/src/main/kotlin/me/lucasbatista/vienna/sdk/entity/ProductVariant.kt
package me.lucasbatista.vienna.sdk.entity

import java.net.URL

class ProductVariant(
    id: String? = null,
    productId: String? = null,
    originalPrice: Double? = null,
    sellingPrice: Double? = null,
    title: String? = null,
    image: URL? = null,
) {
    lateinit var id: String
    lateinit var productId: String
    var originalPrice: Double = 0.0
    var sellingPrice: Double = 0.0
    lateinit var title: String
    lateinit var image: URL

    init {
        id?.let { this.id = id }
        productId?.let { this.productId = productId }
        originalPrice?.let { this.originalPrice = it }
        sellingPrice?.let { this.sellingPrice = it }
        title?.let { this.title = title }
        image?.let { this.image = image }
    }
}

File: mobile/lib/core/entities/product_variant.dart
import 'entity.dart';

class ProductVariant implements Entity {
  late final String id;
  late final double originalPrice;
  late final double sellingPrice;
  late final String title;
  late final Uri image;

  ProductVariant.lazy({
    String? id,
    double? originalPrice,
    double? sellingPrice,
    String? title,
    Uri? image,
  }) {
    if (id != null) this.id = id;
    if (originalPrice != null) this.originalPrice = originalPrice;
    if (sellingPrice != null) this.sellingPrice = sellingPrice;
    if (title != null) this.title = title;
    if (image != null) this.image = image;
  }
}
